tensorflow_probability/python/bijectors/gev_cdf.py

```diff
@@ -0,0 +1,197 @@
+               concentration=0,
+               validate_args=False,
+               name='generalizedextremevalue_cdf'):
+    """Instantiates the `GeneralizedExtremeValueCDF` bijector.
+
+    Args:
+      loc: Float-like `Tensor` that is the same dtype and is
+        broadcastable with `scale` and `concentration`.
+      scale: Positive Float-like `Tensor` that is the same dtype and is
+        broadcastable with `loc` and `concentration`.
+      concentration: Nonzero float-like `Tensor` that is the same dtype and is
+        broadcastable with `loc` and `scale`.
+      validate_args: Python `bool` indicating whether arguments should be
+        checked for correctness.
+      name: Python `str` name given to ops managed by this object.
+    """
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      dtype = dtype_util.common_dtype(
+          [loc, scale, concentration], dtype_hint=tf.float32)
+      self._loc = tensor_util.convert_nonref_to_tensor(
+          loc, dtype=dtype, name='loc')
+      self._scale = tensor_util.convert_nonref_to_tensor(
+          scale, dtype=dtype, name='scale')
+      self._concentration = tensor_util.convert_nonref_to_tensor(
+          concentration, dtype=dtype, name='concentration')
+      super(GeneralizedExtremeValueCDF, self).__init__(
+          validate_args=validate_args,
+          forward_min_event_ndims=0,
+          parameters=parameters,
+          name=name)
+
+  @property
+  def loc(self):
+    """
+    The `loc` in `Y = exp(-t(X))`
+    where `t(x) = (1 + conc * (x - loc) / scale) ) ** (-1 / conc)`
+    when `conc != 0` and `exp(-(x - loc) / scale)` when `conc = 0`.`
+    """
+    return self._loc
+
+  @property
+  def scale(self):
+    """
+    The `scale` in `Y = exp(-t(X))`
+    where `t(x) = (1 + conc * (x - loc) / scale) ) ** (-1 / conc)`
+    when `conc != 0` and `exp(-(x - loc) / scale)` when `conc = 0`.`
+    """
+    return self._scale
+
+  @property
+  def concentration(self):
+    """
+    The `concentration` in `Y = exp(-t(X))`
+    where `t(x) = (1 + conc * (x - loc) / scale) ) ** (-1 / conc)`
+    when `conc != 0` and `exp(-(x - loc) / scale)` when `conc = 0`.`
+    """
+    return self._concentration
+
+  @classmethod
+  def _is_increasing(cls):
+    return True
+
+  def _forward(self, x):
+    with tf.control_dependencies(self._maybe_assert_valid_x(x)):
+      conc = tf.convert_to_tensor(self.concentration)
+      condition = tf.equal(conc, 0.)
+
+      z = (x - self.loc) / self.scale
+      t = tf.where(condition, tf.exp(-z),
+                   tf.exp(-tf.math.log1p(z * conc) / conc))
+      return tf.exp(-t)
+
+  def _inverse(self, y):
+    with tf.control_dependencies(self._maybe_assert_valid_y(y)):
+      t = - tf.math.log(y)
+
+      conc = tf.convert_to_tensor(self.concentration)
+      condition = tf.equal(conc, 0.)
+      z = tf.where(condition, -tf.math.log(t),
+                   tf.math.expm1(-tf.math.log(t) * conc) / conc)
```

Comment:
I wonder if Taylor expanding for small conc is the right thing to do here. That would guarantee stability of this expression

Reply:
I can add a Taylor expansion here. In fact, if we add Taylor expansion, we can add this to all similar places. Also, we can check that using Taylor expansion is in fact using the case conc=0, which is Gumbel bijector as suggested by @jedisom . Thus in my last update, I add a `tol` to control when the bijector converges to the Gumbel bijector.

Reply:
I would avoid a tolerance parameter. I was thinking that there is a radius R where if log(t) and conc < R, then using a taylor expansion would be the right thing to do.
Happy to punt on this for now since it is a bit non-trivial.